tools/server/webui/src/lib/components/app/misc/MarkdownContent.svelte

```diff
@@ -64,136 +58,174 @@
-
-		const tempDiv = document.createElement('div');
-		tempDiv.innerHTML = html;
-
-		// Make all links open in new tabs
-		const linkElements = tempDiv.querySelectorAll('a[href]');
-		let mutated = false;
+	/**
+	 * Removes click event listeners from copy and preview buttons.
+	 * Called on component destroy.
+	 */
+	function cleanupEventListeners() {
+		if (!containerRef) return;
 
-		for (const link of linkElements) {
-			const target = link.getAttribute('target');
-			const rel = link.getAttribute('rel');
+		const copyButtons = containerRef.querySelectorAll<HTMLButtonElement>('.copy-code-btn');
+		const previewButtons = containerRef.querySelectorAll<HTMLButtonElement>('.preview-code-btn');
 
-			if (target !== '_blank' || rel !== 'noopener noreferrer') {
-				mutated = true;
-			}
+		for (const button of copyButtons) {
+			button.removeEventListener('click', handleCopyClick);
+		}
 
-			link.setAttribute('target', '_blank');
-			link.setAttribute('rel', 'noopener noreferrer');
+		for (const button of previewButtons) {
+			button.removeEventListener('click', handlePreviewClick);
 		}
+	}
 
-		return mutated ? tempDiv.innerHTML : html;
+	/**
+	 * Removes this component's highlight.js theme style from the document head.
+	 * Called on component destroy to clean up injected styles.
+	 */
+	function cleanupHighlightTheme() {
+		if (!browser) return;
+
+		const existingTheme = document.getElementById(themeStyleId);
+		existingTheme?.remove();
 	}
 
+	/**
+	 * Enhances code blocks with wrapper, header, language label, and action buttons.
+	 * Adds copy button to all code blocks and preview button to HTML blocks.
+	 * @param html - The HTML string containing code blocks to enhance
+	 * @returns Enhanced HTML string with wrapped code blocks
+	 */
 	function enhanceCodeBlocks(html: string): string {
-		if (!html.includes('<pre')) {
-			return html;
-		}
+		return processHtml(html, '<pre', (tempDiv) => {
+			const preElements = tempDiv.querySelectorAll('pre');
+			let mutated = false;
 
-		const tempDiv = document.createElement('div');
-		tempDiv.innerHTML = html;
+			for (const [index, pre] of Array.from(preElements).entries()) {
+				const codeElement = pre.querySelector('code');
 
-		const preElements = tempDiv.querySelectorAll('pre');
-		let mutated = false;
+				if (!codeElement) {
+					continue;
+				}
 
-		for (const [index, pre] of Array.from(preElements).entries()) {
-			const codeElement = pre.querySelector('code');
+				mutated = true;
 
-			if (!codeElement) {
-				continue;
-			}
+				let language = 'text';
+				const classList = Array.from(codeElement.classList);
 
-			mutated = true;
+				for (const className of classList) {
+					if (className.startsWith('language-')) {
+						language = className.replace('language-', '');
+						break;
+					}
+				}
 
-			let language = 'text';
-			const classList = Array.from(codeElement.classList);
+				const rawCode = codeElement.textContent || '';
+				const codeId = `code-${uuid()}-${index}`;
```

Comment:
same remark here for `uuid()`

Reply:
a10adb5674bc0f67bf1ef2862386b221b1397120